blog.py

```diff
@@ -322,15 +322,18 @@ def _is_ess_empty(self, lines: List[str]) -> int:
         content = ''.join(line.strip() for line in lines)
         is_empty = re.sub(r'\*\*.+\*\*', '', content).replace('**', '') == ''
         return 1 if is_empty else 0
-  
-    def delete(self, posts: Union[str, List[str]]):
+
+    def trash(self, posts: Union[str, List[str]]):
         if isinstance(posts, str):
             posts = [posts]
-        for file in posts:
-            os.remove(file)
+        path_ = os.path.join(BASE_DIR, 'trash')
+        if not os.path.isdir(path_):
+            os.mkdir(path_)
+        for post in posts:
+            shutil.move(post, path_)
         qmark = ', '.join(['?'] * len(posts))
         sql = f'DELETE FROM posts WHERE path in ({qmark})'
-        self._conn.execute(sql, posts)
+        self._conn.execute(sql, posts) 
```

Comment:
remove the trailing spaces

Formatting is a tedious work that shouldn't be done manually. You should try to use IDE to help you format code or use command line tools such as `yapf` to format the code for you. It makes sure that your code is formatted consistently.